### PR TITLE
Add __all__ tags to modules in package apache_beam/testing

### PR DESCRIPTION
**Pull request by charlesccychen**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [ ] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [ ] Make sure tests pass via `mvn clean verify`.
 - [ ] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [ ] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

**Comment by charlesccychen** [comment]
R: @robertwb

File: sdks/python/apache_beam/testing/test_utils.py [hunk 1]
```diff
@@ -23,6 +23,13 @@
 
 from apache_beam.utils import retry
 
+
+__all__ = [
+    'compute_hash',
```
Comment (robertwb) [inline sdks/python/apache_beam/testing/test_utils.py]: Are these really intended to be part of the public API?

Comment (charlesccychen) [inline sdks/python/apache_beam/testing/test_utils.py]: What is the bar we are using for "public" and inclusion in __all__?  This is public to this module in the sense that tests can use this, but likely doesn't need to be public to users.

**Comment by charlesccychen** [comment]
Also, assert_that and associated matchers are in https://github.com/apache/beam/blob/e2be6961d07f1c7efe73dbc00280c8e5c9c94dfa/sdks/python/apache_beam/transforms/util.py -- are you suggesting a move into testing/?

**Comment by robertwb** [comment]
Yes, I thought we were intending on moving these into testing.

**Comment by charlesccychen** [comment]
Thanks, PTAL--added "For internal use only; no backwards-compatibility guarantees." per the `__all__` use guidance.

File: sdks/python/apache_beam/testing/test_utils.py (hunk 1 above)
Comment (robertwb) [inline sdks/python/apache_beam/testing/test_utils.py]: The bar is public to users.

**Pull request closed without merging**